src/service/allocator/allocator.service.ts

```diff
@@ -0,0 +1,342 @@
+    const averageSuccessRate =
+      await this.prismaService.allocators_weekly.aggregate({
+        _avg: {
+          avg_weighted_retrievability_success_rate: true,
+        },
+      });
+
+    const allocatorCountResult = await this.prismaService.$queryRaw<
+      [
+        {
+          count: number;
+        },
+      ]
+    >`select count(distinct allocator)::int
+      from client_allocator_distribution_weekly`;
+
+    const weeklyHistogramResult =
+      await this.histogramHelper.getWeeklyHistogramResult(
+        await this.prismaService.$queryRawTyped(getAllocatorRetrievability()),
+        allocatorCountResult[0].count,
+      );
+
+    return RetrievabilityWeekResponseDto.of(
+      averageSuccessRate._avg.avg_weighted_retrievability_success_rate * 100,
+      weeklyHistogramResult,
+    );
+  }
+
+  async getAllocatorBiggestClientDistribution() {
+    const allocatorCountResult = await this.prismaService.$queryRaw<
+      [
+        {
+          count: number;
+        },
+      ]
+    >`select count(distinct allocator)::int
+      from client_allocator_distribution_weekly`;
+
+    return await this.histogramHelper.getWeeklyHistogramResult(
+      await this.prismaService.$queryRawTyped(
+        getAllocatorBiggestClientDistribution(),
+      ),
+      allocatorCountResult[0].count,
+    );
+  }
+
+  async getAllocatorSpsCompliance(): Promise<SpsComplianceWeekResponseDto> {
+    const weeks = await this.prismaService.providers_weekly
+      .findMany({
+        distinct: ['week'],
+        select: {
+          week: true,
+        },
+      })
+      .then((r) => r.map((p) => p.week));
+
+    const allocatorCount = (
+      await this.prismaService.allocators_weekly.findMany({
+        distinct: ['allocator'],
+        select: {
+          allocator: true,
+        },
+      })
+    ).length;
+
+    const calculationResults: {
+      results: {
+        upToOneCompliantProvidersPercent: number;
+        twoCompliantProvidersPercent: number;
+        threeCompliantProvidersPercent: number;
+      }[];
+      week: Date;
+    }[] = [];
+    for (const week of weeks) {
+      const thisWeekAverageRetrievability =
+        await this.prismaService.providers_weekly.aggregate({
+          _avg: {
+            avg_retrievability_success_rate: true,
+          },
+          where: {
+            week: week,
+          },
+        });
+
+      const weekProviders = await this.prismaService.providers_weekly.findMany({
+        where: {
+          week: week,
+        },
+      });
+
+      const weekProvidersCompliance = weekProviders.map((wp) => {
+        let complianceScore = 0;
+        if (
+          wp.avg_retrievability_success_rate >
+          thisWeekAverageRetrievability._avg.avg_retrievability_success_rate
+        )
+          complianceScore++;
+        if (wp.num_of_clients > 3) complianceScore++;
+        if (
+          wp.biggest_client_total_deal_size * 100n <=
+          33n * wp.total_deal_size
```

Comment:
```suggestion
          30n * wp.total_deal_size
```
We want 30% threshold